pkgs/development/tools/comby/default.nix

```diff
@@ -0,0 +1,50 @@
+{ pkgs, ocamlPackages, fetchFromGitHub, lib }:
+let
+  mkCombyPackage = { pname }:
+    ocamlPackages.buildDunePackage rec {
+      inherit pname;
+      version = "1.5.1";
+      useDune2 = true;
+      minimumOcamlVersion = "4.11.0";
+
+      src = fetchFromGitHub {
+        owner = "comby-tools";
+        repo = "comby";
+        rev = version;
+        sha256 = "1ipfrr6n1jyyryhm9zpn8wwgzfac1zgbjdjzrm00qcwc17r8x2hf";
+      };
+
+      buildInputs = [
+        pkgs.zlib
+        pkgs.pkg-config
+        ocamlPackages.shell
+        ocamlPackages.ppx_deriving
+        ocamlPackages.ppx_deriving_yojson
+        ocamlPackages.ppx_sexp_conv
+        ocamlPackages.ppx_jane
+        ocamlPackages.ppx_expect
+        ocamlPackages.dune-configurator
+        ocamlPackages.lwt
+        ocamlPackages.patience_diff
+        ocamlPackages.core
+        ocamlPackages.toml
+        ocamlPackages.cohttp-lwt-unix
+        ocamlPackages.mparser
+        ocamlPackages.opium
+        ocamlPackages.textutils
+        ocamlPackages.jst-config
+        ocamlPackages.parany
+        ocamlPackages.mparser-pcre
+        ocamlPackages.ocaml_pcre
+      ];
+
+      meta = {
+        description = "Tool for searching changing code structure";
+        license = lib.licenses.asl20;
+        homepage = "https://comby.dev/";
+      };
+    };
+
+  combyKernel = mkCombyPackage { pname = "comby-kernel"; };
+in (mkCombyPackage { pname = "comby"; }).overrideAttrs
+(oldAttrs: { buildInputs = oldAttrs.buildInputs ++ [ combyKernel ]; })
```

Comment:
Consider adding an `extraBuildInputs` parameter rather than overriding the derivation here.